exercise_1/task_1.cpp

```diff
@@ -0,0 +1,39 @@
+#include <iostream>
+#include <string>
+#include <cctype>
+
+using namespace std;
+
+int main()
+{
+	string words;	
+	int vowels = 0, consonants = 0, others = 0;
+	
+	cout << "Enter words (q to quit)\n";
+	while ((cin >> words)&&(words != "q"))
+	{
+		if (isalpha(words[0]))
+		{
+			switch (tolower(words[0])) {
+			case 'a':
+			case 'i':
+			case 'e':
+			case 'o':
+			case 'u':
+			case 'y':
+				vowels++;
+				break;
+			default:
+				consonants++;
+				break;
+			}
+		}
+		else {
+			others++;
+		}
+	}
+  
+	cout << vowels << " words beginning with vowel\n";
+	cout << consonants << " words beginning with consonant\n";
+	cout << others << " others";
+}
```

Comment:
main() is probably the only non-void function in C++ that is not complaining about missing return statement.
From C++14 standard:
> A return statement in main has the effect of leaving the main function (destroying any objects with automatic
> storage duration) and calling std::exit with the return value as the argument. If control reaches the end
> of main without encountering a return statement, the effect is that of executing
> return 0;

However, it is good to remember about `return 0;` or better `return EXIT_SUCCESS;` at the end of main function.
Why EXIT_SUCCESS or EXIT_FAILURE is better, then just returning '0' or '1'? Portability reasons :)

Reply:
Hi, when should I merge a pull request?

Reply:
I've also added one commit within this pull request below and not sure if you saw it.